### PR TITLE
DOC-1535: Remove limitations of computed columns in FKs

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes DOC-1535

### CODE REVIEW

File: v21.1/computed-columns.md [hunk 1]
```diff
@@ -28,7 +27,6 @@ Virtual computed columns:
 - Are not stored in the table's primary index.
 - Are recomputed as the column data in the expression changes.
 - Cannot be used as part of a primary key, `FAMILY` definition, in `CHECK` constraints, or `FOREIGN KEY` constraints.
-- Cannot be a [foreign key](foreign-key.html) reference.
```
Comment (mgartner) [inline v21.1/computed-columns.md:31]: This restriction still remains. It's quite confusing and I had to double check myself. Here are examples of the differences:

A computed column expression can now reference a foreign key column, so this is allowed:

```sql
defaultdb> CREATE TABLE parent (p INT PRIMARY KEY);
CREATE TABLE

defaultdb> CREATE TABLE child (c INT PRIMARY KEY, c_p INT REFERENCES parent (p), v INT AS (c_p + 10) VIRTUAL);
CREATE TABLE
```

But this is still NOT allowed:

```sql
defaultdb> CREATE TABLE parent (p INT PRIMARY KEY, v INT AS (p + 10) VIRTUAL, UNIQUE INDEX (v))
CREATE TABLE

defaultdb> CREATE TABLE child (c INT PRIMARY KEY, v INT REFERENCES parent (v))
ERROR: unimplemented: virtual column "v" cannot be referenced by a foreign key
SQLSTATE: 0A000
HINT: You have attempted to use a feature that is not yet implemented.
See: https://go.crdb.dev/issue-v/59671/v22.1
```

File: v21.1/foreign-key.md [hunk 2]
```diff
@@ -26,7 +26,6 @@ To read more about how foreign keys work, see our [What is a Foreign Key? (With
 **Foreign Key Columns**
 
 - Foreign key columns must use their referenced column's [type](data-types.html).
-- A foreign key column cannot be a [computed column](computed-columns.html).
```
Comment (mgartner) [inline v21.1/foreign-key.md:29]: This should read: "A foreign key column cannot be a [virtual computed column](...)" (see my second example above). You might want to clarify that a foreign key column **can** be a stored computed column, though.

File: v21.2/computed-columns.md [hunk 3]
```diff
@@ -36,7 +35,6 @@ Virtual computed columns:
 - Are not stored in the table's primary index.
 - Are recomputed as the column data in the expression changes.
 - Cannot be used as part of a primary key, `FAMILY` definition, in `CHECK` constraints, or `FOREIGN KEY` constraints.
-- Cannot be a [foreign key](foreign-key.html) reference.
```
Comment (mgartner) [inline v21.2/computed-columns.md:39]: ditto: leave as-is

File: v21.2/foreign-key.md [hunk 4]
```diff
@@ -26,7 +26,6 @@ To read more about how foreign keys work, see our [What is a Foreign Key? (With
 **Foreign Key Columns**
 
 - Foreign key columns must use their referenced column's [type](data-types.html).
-- A foreign key column cannot be a [computed column](computed-columns.html).
```
Comment (mgartner) [inline v21.2/foreign-key.md:29]: ditto: see my previous comment for the same line in 21.1 docs

**Review by mgartner — CHANGES_REQUESTED** [review]

File: v21.1/computed-columns.md (hunk 1 above)
Comment (nickvigilante) [reply v21.1/computed-columns.md:31]: Oof, that's tricky. Took me a while to wrap my head around it, but I think I understand. Basically, you can create a computed column that references a foreign key, but you can't create a foreign key that references a computed column.

Done.

File: v21.2/computed-columns.md (hunk 3 above)
Comment (nickvigilante) [reply v21.2/computed-columns.md:39]: Done.

File: v21.2/foreign-key.md (hunk 4 above)
Comment (nickvigilante) [reply v21.2/foreign-key.md:29]: Done.

**Review by mgartner — APPROVED** [review]

**Review by lnhsingh — COMMENTED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12817#-:-MuDguyi18qRuqsMxjex:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gemma-shay)

**Review by gemma-shay — APPROVED** [review]

**Pull request merged**